supabase/migrations/20250820220302_fix-office-hours.sql

```diff
@@ -0,0 +1,1358 @@
+
+revoke truncate on table "public"."notification_preferences" from "anon";
+
+revoke update on table "public"."notification_preferences" from "anon";
+
+revoke delete on table "public"."notification_preferences" from "authenticated";
+
+revoke insert on table "public"."notification_preferences" from "authenticated";
+
+revoke references on table "public"."notification_preferences" from "authenticated";
+
+revoke select on table "public"."notification_preferences" from "authenticated";
+
+revoke trigger on table "public"."notification_preferences" from "authenticated";
+
+revoke truncate on table "public"."notification_preferences" from "authenticated";
+
+revoke update on table "public"."notification_preferences" from "authenticated";
+
+revoke delete on table "public"."notification_preferences" from "service_role";
+
+revoke insert on table "public"."notification_preferences" from "service_role";
+
+revoke references on table "public"."notification_preferences" from "service_role";
+
+revoke select on table "public"."notification_preferences" from "service_role";
+
+revoke trigger on table "public"."notification_preferences" from "service_role";
+
+revoke truncate on table "public"."notification_preferences" from "service_role";
+
+revoke update on table "public"."notification_preferences" from "service_role";
+
+alter table "public"."notification_preferences" drop constraint "notification_preferences_class_id_fkey";
+
+alter table "public"."notification_preferences" drop constraint "notification_preferences_user_id_fkey";
+
+alter table "public"."notification_preferences" drop constraint "notification_preferences_pkey";
+
+drop index if exists "public"."notification_preferences_pkey";
+
+drop table "public"."notification_preferences";
+
+drop type "public"."email_digest_frequency";
+
+drop type "public"."notification_type";
+
+set check_function_bodies = off;
+
+CREATE OR REPLACE FUNCTION public.assignment_before_update()
+ RETURNS trigger
+ LANGUAGE plpgsql
+ SECURITY DEFINER
+AS $function$
+BEGIN
+   CASE TG_OP
+   WHEN 'UPDATE' THEN
+      IF OLD.template_repo is not null and OLD.template_repo != NEW.template_repo then
+         UPDATE autograder_regression_test SET repository = NEW.template_repo WHERE repository = OLD.template_repo AND autograder_id = NEW.id;
+      elseif OLD.template_repo is null AND NEW.template_repo is not null then
+         INSERT INTO autograder_regression_test (repository, autograder_id) VALUES (NEW.template_repo, NEW.id);
+      elseif OLD.template_repo is not null and NEW.template_repo is null then
+         DELETE FROM autograder_regression_test WHERE repository = NEW.template_repo and autograder_id = NEW.id;
+      end if;
+      RETURN NEW;
+   ELSE
+      RAISE EXCEPTION 'Unexpected TG_OP: "%". Should not occur!', TG_OP;
+   END CASE;
+   
+END
+$function$
+;
+
+CREATE OR REPLACE FUNCTION public.authorizeforinstructorofstudent(_user_id uuid)
```

Comment:
What is supposed to have changed about this function? This introduces a performance regression.